azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/DiskApiLiveTest.java

```diff
@@ -0,0 +1,99 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+
+import java.net.URI;
+import java.util.List;
+
+import org.jclouds.azurecompute.arm.domain.CreationData;
+import org.jclouds.azurecompute.arm.domain.Disk;
+import org.jclouds.azurecompute.arm.domain.DiskProperties;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiLiveTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.common.base.Predicate;
+
+import static org.jclouds.util.Predicates2.retry;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+@Test(groups = "live", singleThreaded = true)
+public class DiskApiLiveTest extends BaseAzureComputeApiLiveTest {
+
+   private String diskName;
+
+   @BeforeClass
+   @Override
+   public void setup() {
+      super.setup();
+      createTestResourceGroup();
+      diskName = "jclouds-" + RAND;
+   }
+
+   @Test
+   public void deleteDiskResourceDoesNotExist() {
+      assertNull(api().delete(diskName));
+   }
+
+   @Test(dependsOnMethods = "deleteDiskResourceDoesNotExist")
+   public void createDisk() {
+      DiskProperties properties = DiskProperties.builder().creationData(CreationData.create("Empty")).diskSizeGB(2).build();
+      Disk dataDisk = api().createOrUpdate(diskName, LOCATION, properties);
+      assertTrue(stateReached("Succeeded"), "start operation did not complete in the configured timeout");
+      assertTrue(dataDisk.properties().diskSizeGB() == 2);
+   }
+
+   @Test(dependsOnMethods = "createDisk")
+   public void getDisk() {
+      Disk dataDisk = api().get(diskName);
+      assertNotNull(dataDisk.name());
+      assertTrue(dataDisk.properties().diskSizeGB() == 2);
+   }
+
+   @Test(dependsOnMethods = "createDisk")
+   public void listDisks() {
+      List<Disk> dataDisks = api().list();
+      assertTrue(dataDisks.size() > 0);
+   }
+
+   @Test(dependsOnMethods = {"listDisks", "getDisk"}, alwaysRun = true)
+   public void deleteDisk() {
+      URI uri = api().delete(diskName);
+      assertNotNull(uri);
+   }
+
+   private DiskApi api() {
+      return api.getDiskApi(resourceGroupName);
+   }
+
+   private boolean waitForState(String name, final String state) {
+      return api().get(name).properties().provisioningState().equals(state);
+   }
+
+   private boolean stateReached(final String state) {
+      return retry(new Predicate<String>() {
+         @Override
+         public boolean apply(String name) {
+            return waitForState(name, state);
+         }
+      }, 60 * 4 * 1000).apply(diskName);
+   }
```

Comment:
Better change these two methods to use the already available predicate to poll Provisionable instances:

```java
 private boolean waitUntilAvailable(String name) {
   return resourceAvailable.apply(new Supplier<Provisionable>() {
      @Override public Provisionable get() {
           Disk disk api().get(name);
           return disk == null ? null : disk.properties();
      }
   });
}
```